src/components/clusterWizard/ClusterWizardForm.tsx

```diff
@@ -17,42 +18,42 @@ import {
   Spinner,
 } from '@patternfly/react-core';
 import { ExternalLinkAltIcon } from '@patternfly/react-icons';
-import { useDispatch } from 'react-redux';
 
-import ClusterWizardToolbar from './ClusterWizardToolbar';
 import PageSection from '../ui/PageSection';
 import { ToolbarButton, ToolbarText } from '../ui/Toolbar';
 import { InputField, TextAreaField, SelectField } from '../ui/formik';
-import validationSchema from './validationSchema';
 import GridGap from '../ui/GridGap';
-import { Cluster, ClusterUpdateParams } from '../../api/types';
+import { ClusterUpdateParams } from '../../api/types';
 import { WizardStep } from '../../types/wizard';
 import { patchCluster } from '../../api/clusters';
 import { handleApiError } from '../../api/utils';
 import { OPENSHIFT_VERSION_OPTIONS, CLUSTER_MANAGER_SITE_LINK } from '../../config/constants';
 import AlertsSection from '../ui/AlertsSection';
 import { updateCluster } from '../../features/clusters/currentClusterSlice';
+import { selectCurrentCluster } from '../../selectors/currentCluster';
+import ClusterWizardToolbar from './ClusterWizardToolbar';
+import validationSchema from './validationSchema';
 
 interface ClusterWizardFormProps {
-  cluster: Cluster;
   setStep: React.Dispatch<React.SetStateAction<WizardStep>>;
 }
 
-const ClusterWizardForm: React.FC<ClusterWizardFormProps> = ({ cluster, setStep }) => {
+const ClusterWizardForm: React.FC<ClusterWizardFormProps> = ({ setStep }) => {
   const dispatch = useDispatch();
+  const cluster = useSelector(selectCurrentCluster);
```

Comment:
Hmm, having to consider cluster being undefined is quite bad. We're already making sure that cluster is set on ClusterPage so we can be sure that cluster is always set in this component and also in ClusterWizard. I think a better solution is to keep passing the cluster as prop as it was before. Alternatively we can use `connect` but I think using props here is better.

Reply:
You are right, my intention was to make the flow simpler but actually it not become truth, I am removing the last commit.